doc/connectivity/usb/device_next/usb_device.rst

```diff
@@ -76,3 +76,112 @@ configuration ``-DCONF_FILE=usbd_next_prj.conf`` either directly or via
+   :end-before: doc string instantiation end
+
+USB device requires at least one configuration. If the USB controller assigned
+to the device context supports high speed, it needs at least one configuration
+instance per speed. The application should use
+:c:macro:`USBD_CONFIGURATION_DEFINE` to instantiate a configuration. Later, USB
+device functions are assigned to a configuration.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc configuration instantiation start
+   :end-before: doc configuration instantiation end
+
+Any string descriptor must be added to the device context at runtime before
+initializing the USB device with :c:func:`usbd_add_descriptor`.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc add string descriptor start
+   :end-before: doc add string descriptor end
+
+Each configuration instance for a specific speed must be added to the device
+context at runtime before the USB device is initialized using
+:c:func:`usbd_add_configuration`. See :c:enumerator:`USBD_SPEED_FS` and
+:c:enumerator:`USBD_SPEED_HS`. For each of the spedific speed configuration
+number is assigned based on the order of addition to the context, means the
+first full speed configuration will get number one, so called high speed, and
+further accordingly.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc configuration register start
+   :end-before: doc configuration register end
+
+
+Although we have already done a lot, this USB device has no function. A device
+can have multiple configuration with different set of function, event at
+different speed. A function or class can be registered on a USB device before
+it is initialized using :c:func:`usbd_register_class`. The specific
+configuration is specified using :c:enumerator:`USBD_SPEED_FS` or
+:c:enumerator:`USBD_SPEED_HS` and the configuration number.  For simple cases,
+:c:func:`usbd_register_all_classes` can be used to register all available
+instances.
+
+.. literalinclude:: ../../../../samples/subsys/usb/common/sample_usbd_init.c
+   :language: c
+   :start-after: doc functions register start
+   :end-before: doc functions register end
+
+The last step in the preparation is to initialize the device with
+:c:func:`usbd_init`. After this, the configuration of the device cannot be
+changed. A device can be deinitialized with :c:func:`usbd_shutdown` and all
+instances can be reused, but the previous steps must be repeated. So it is
+possible to shutdown a device, register another type of configuration or
+function, and initialize it again.  At the USB controller level,
+:c:func:`usbd_init` does only what is necessary to detect VBUS changes. There
+are controller types where the next step is only possible if a VBUS signal is
+present, more on this later.
```

Comment:
```suggestion
present.
```

The "more on this later" only adds confusion and seem completely unnecessary here.